zerver/models.py

```diff
@@ -1009,7 +1007,13 @@ def get_active_streams(realm):
 
 def get_stream(stream_name, realm):
     # type: (Text, Realm) -> Stream
-    return get_stream_backend(stream_name, realm.id)
+    '''
+    We eventually want get_stream to take a realm_id, so
+    that callers don't need to fetch a Realm object for no
+    reason.  Until we do that code sweep, callers that care
+    about performance should call get_realm_stream directly.
+    '''
```

Comment:
For this, do we actually want to call this with the ID everywhere?  It seems potentially fine to just offer both interfaces; I imagine at least if you're working on a management shell, you probably want the `get_stream` interface since it's less annoying.

Reply:
The goal is to rename it back to get_realm after the massive code sweep (mostly tests).  I'm actually fine to drop the two get_realm_stream commits from this PR.  The context here was I was trying to eliminate queries, but I misdiagnosed the one caller as contributing to an extra DB hit.  I left the commit in since it's harmless and could eventually help with query counts.